loadbalancer/algorithm_test.go

```diff
@@ -312,6 +312,58 @@ func TestConsistentHashSearch(t *testing.T) {
 	}
 }
 
+func TestConsistentHashBoundedLoadSearch(t *testing.T) {
+	endpoints := []string{"http://127.0.0.1:8080", "http://127.0.0.2:8080", "http://127.0.0.3:8080"}
+	ch := newConsistentHash(endpoints).(consistentHash)
+
+	r, _ := http.NewRequest("GET", "http://127.0.0.1:1234/foo", nil)
+	r.RemoteAddr = "192.168.0.1:8765"
+	route := NewAlgorithmProvider().Do([]*routing.Route{{
+		Route: eskip.Route{
+			BackendType: eskip.LBBackend,
+			LBAlgorithm: ConsistentHash.String(),
+			LBEndpoints: endpoints,
+		},
+	}})[0]
+	ctx := &routing.LBContext{Request: r, Route: route, Params: map[string]interface{}{ConsistentHashKey: net.RemoteHost(r).String()}}
+	noLoad := ch.boundedLoadSearch("somekey", 1.25, ctx)
+	nonBounded := ch.search("somekey")
+
+	if noLoad != nonBounded {
+		t.Error("When no endpoints are overloaded, the chosen endpoint should be the same as standard consistentHash")
+	}
+	// now we know that noLoad is the endpoint which should be requested for somekey if load is not an issue.
+	for range [20]int{} {
+		route.LBEndpoints[noLoad].Metrics.IncInflightRequest()
+	}
```

Comment:
Should we introduce a reusbale helper function for this like
```go
addInflightRequests(route.LBEndpoints[noLoad], 20)
```

Reply:
Is there a convention on where to put helper functions for tests? Or can I just put it as a private method in algorithm_test.go